### ISSUE TITLE
Date formatting is inconsistent when using English language

### ISSUE THREAD

**Issue by MikelThief** [ArturB/WUT-Thesis#35]
I stumbled upon inconsistency issue related to date formats. By creating a bibliography entry 

```
@online{merge,
    title = "{Merging a pull request}",
    date = {2017-07-28},
    organization = "Github",
    author = "{John Smith}",
    url = {https://www.differential.com/posts/best-way-to-merge-a-github-pull-request},
    urldate = {2020-01-27}
}
```

I receive inconsistent visual representation:
![incorrect date format](https://user-images.githubusercontent.com/8342147/73224345-819ad180-4169-11ea-9be9-e817e02ca4be.png)

To be consistent with format required by the norm (i.e. DD.MM.YYYY) I set up following piece of code to override short and long date formats:

```
\DefineBibliographyExtras{english}{%
  \protected\def\mkbibdatelong#1#2#3{%
    \iffieldundef{#3}
      {}
      {\mkdayzeros{\thefield{#3}}\adddot
       \iffieldundef{#2}{}{\thinspace}}%
    \iffieldundef{#2}
      {}
      {\mkmonthzeros{\thefield{#2}}%
       \iffieldundef{#1}
     {}
     {\iffieldundef{#3}{/}{\adddot\thinspace}}}%
    \iffieldbibstring{#1}
      {\bibstring{\thefield{#1}}}
      {\dateeraprintpre{#1}\mkyearzeros{\thefield{#1}}}}%
  \protected\def\mkbibdateshort#1#2#3{%
    \iffieldundef{#3}
      {}
      {\mkdayzeros{\thefield{#3}}\adddot
       \iffieldundef{#2}{}{\thinspace}}%
    \iffieldundef{#2}
      {}
      {\mkmonthzeros{\thefield{#2}}%
       \iffieldundef{#1}
     {}
     {\iffieldundef{#3}{/}{\adddot\thinspace}}}%
    \iffieldbibstring{#1}
      {\bibstring{\thefield{#1}}}
      {\dateeraprintpre{#1}\mkyearzeros{\thefield{#1}}}}%
}
```
With the fix above I get:
![correct date format](https://user-images.githubusercontent.com/8342147/73224073-9c207b00-4168-11ea-90ec-f16ea67012fb.png)

Behaviour was spotted using Overleaf service.

I will be happy to create a PR to fix those issues.

**Comment by ArturB** [comment]
Well, this is rather minor issue (I suppose that majority of supervisors would not even notice it), and the code you pasted is somewhat complicated, and need to be tested before merging to master As for now, please stick to your own fix in your own thesis, and I'll deal with it later (that is after 14 Feb, because I also have to finish my master's thesis, just like others, by this day).

**Comment by MikelThief** [comment]
I disagree that this is minor issue. You certainly spent not enough time with our beloved secretary of Intitute of CS who cares a lot about such thing ;)
I wouldn't say that this code is complicated. We are all using it already. It is just hidden under default settings. I don't submit my thesis this February and I am comfortable with waiting. Let me know when you are ready to merge the PR - I will try to prepare it shortly after your ping.

**Comment by pwozniak** [comment]
Hi, co-author of this project here,

Firstly, thank you for your issue.

I have checked the other possibilities (\DeclareSourceMap etc.) and after few days of research I am sure that's your solution is the only one available. Biblatex package doesn't support "dd.mm.yyyy" formatting in BIB files.

All the date modifications on *.bib level results with the warning:
```
WARN - Entry 'merge' (bibliografia.bib): Invalid format '{27.01.2020}' of date field 'urldate' - ignoring
```

Biblatex biber engine supports only those date formats:
![](https://user-images.githubusercontent.com/12483609/73607857-03508c00-45bc-11ea-9199-328f66db69d8.png)

# Alternative solution
**However**, you can also use a "trade-off" solution.
Instead of ```@online```, use ```@misc``` just like @ArturB did.
```
@misc{dcp19,
    author       = "Brodzki, Artur M.",
    title        = "{I}mplementation of own steganography protocol {DCP}-19, loosely based on {HICCUPS}",
    howpublished = "Dostęp zdalny (14.03.2019): \url{https://github.com/ArturB/DCP-19}",
    year         = "2019"
}
``` 
In that case, dates aren't parsed by biber. They are treated as plaintext.

# Next steps
To sum up, you have two options to choose.

If you think it is necessary, try to clean up your code first (fix indents and spaces) and then **create the pull request**. If it's possible, fit those lines of code into ```\langeng``` definition.

If you are satisfied with the ```@misc``` trade-off or you would be in a hurry few months later, I think that will pass in your institute's secretary.

Thank you once again and good luck. :)
